### PR TITLE
Consolidate duplicated server traversal functions into clusterHelpers (#80)

### PR DESCRIPTION
**Pull request by dpage**
## Summary

- Consolidate six duplicated server/selection traversal functions
  into the existing `client/src/utils/clusterHelpers.ts` utility
  module
- Add five new exported functions (`extractEstateServerIds`,
  `extractClusterServerIds`, `computeEstateServerCounts`,
  `countEstateServers`) plus `ServerCounts` interface
- Replace local duplicate implementations in six consumer files
  with imports from the shared utility
- Net reduction of ~140 lines of duplicated code

This is the first of several PRs addressing issue #80 (React
client structural improvements). It targets the "server traversal
deduplication" category.

## Changes

| File | Change |
|------|--------|
| `utils/clusterHelpers.ts` | Add 5 new functions + `ServerCounts` interface |
| `utils/__tests__/clusterHelpers.test.ts` | Add 24 new tests (37 total) |
| `EstateDashboard/index.tsx` | Replace `extractAllServerIds` with import |
| `ClusterDashboard/index.tsx` | Replace `extractServerIds` with import |
| `HealthOverviewSection.tsx` | Replace `computeServerCounts` + local `ServerCounts` with imports |
| `KpiTilesSection.tsx` | Replace `countServers` with import |
| `ClusterCardsSection.tsx` | Replace `collectAllServers` with existing `collectServers` import |
| `usePerformanceSummary.ts` | Replace local `extractEstateServerIds` with import |

## Test plan

- [x] All 37 clusterHelpers unit tests pass (100% line coverage)
- [x] All 257 Dashboard/StatusPanel component tests pass
- [x] All 229 context tests pass
- [x] All 182 hook tests pass
- [x] All 316 util tests pass
- [x] ESLint passes on all changed files
- [x] No behavior changes — pure refactor

Closes partially #80

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by CodeRabbit

* **Refactor**
  * Consolidated duplicated server traversal and counting logic into shared utilities used across dashboard and status components; behavior preserved and UI consumption unchanged.

* **Tests**
  * Added comprehensive tests for server extraction, deduplication, counting, and status categorization, covering nested and edge-case hierarchies.

* **Documentation**
  * Added design and planning documents describing the refactor and a multi-phase plan to decompose oversized server and tooling source files.

### CODE REVIEW

File: .claude/specs/2026-04-22-server-traversal-dedup-design.md [hunk 1]
```diff
@@ -0,0 +1,245 @@
...
+```typescript
+export const extractClusterServerIds = (
+    selection: Record<string, unknown>
+): number[] => {
+    const servers = selection.servers as
+        ServerLike[] | undefined;
+    if (!servers) { return []; }
+    return collectServers(servers).map(s => s.id);
+};
+```
```
[the remark this answers is not in the document]
Comment (dpage) [reply .claude/specs/2026-04-22-server-traversal-dedup-design.md:94]: Good catch. Updated the spec snippet to match the actual implementation with Set-based deduplication in 32d37ac.
